spec/date.html

```diff
@@ -546,6 +552,29 @@ <h1>ValidateDate ( _y_, _m_, _d_ )</h1>
       </emu-alg>
     </emu-clause>
 
+    <emu-clause id="sec-temporal-constraindate" aoid="ConstrainDate">
+      <h1>ConstrainDate ( _year_, _month_, _day_ )</h1>
+      <emu-alg>
+        1. Assert: _year_, _month_, and _day_ are integer Number values.
+        1. Set _year_ to ConstrainToRange(_year_, -999999, 999999).
+        1. Set _month_ to ConstrainToRange(_month_, 1, 12).
+        1. Set _day_ to ConstrainToRange(_day_, 1, DaysInMonth(_year_, _month_));
+        1. Return the Record {
+          [[Year]]: _year_,
+          [[Month]]: _month_,
+          [[Day]]: _day_,
+          }.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-balancedate" aoid="BalanceDate">
```

Comment:
I have a BalanceDate draft in https://github.com/tc39/proposal-temporal/pull/333/commits/e79b53a9efc8e597bb1728f24b2f86640c704718 which is in my pull request #333, consider cherry picking that commit into this pull request if it looks reasonable?